src/ir/possible-contents.h

```diff
@@ -572,8 +578,9 @@ template<> struct hash<wasm::NullLocation> {
 
 template<> struct hash<wasm::ConeReadLocation> {
   size_t operator()(const wasm::ConeReadLocation& loc) const {
-    return std::hash<std::pair<wasm::HeapType, wasm::Index>>{}(
-      {loc.type, loc.index});
+    return std::hash<
+      std::pair<wasm::HeapType, std::pair<wasm::Index, wasm::Index>>>{}(
+      {loc.type, {loc.depth, loc.index}});
```

Comment:
Sounds good. I can do that as a followup as it can be done in the other places here that have >2 items?

Reply:
Looking into this, I don't think I know how to do it... just specializing `hash<T>` on tuples and trying to use SFINAE doesn't work:
```cpp
/home/azakai/Dev/binaryen/src/support/hash.h:65:29: error: partial specialization ‘struct std::hash<_Tp>’ does not specialize any template arguments; to define the primary template, remove the template argument list
   65 | template<typename T> struct hash<T> {
      |                             ^~~~~~~
```
Seems like that approach can't work anyhow since `std::get` on a tuple needs the [index to be known at compile time](https://en.cppreference.com/w/cpp/utility/tuple/get), and I was going to do a loop.

Reading some docs like [this](https://stackoverflow.com/questions/16387354/template-tuple-calling-a-function-on-each-element), I'm still not sure how to do it, but even if it's possible those look quite complex. Perhaps overkill for something as simple as this? Or am I missing something obvious?

Reply:
Yeah, you'd have to recurse over the parameter pack at the type level. I can take a look at it if you want. What other locations did you have in mind that could use this?

Reply:
Just some other places in this file. But actually they all use 3 parameters. Maybe I'll just implement that for now, let me take a look there.

Reply:
#5162